icicle/primitives/extension_field.cuh

```diff
@@ -142,8 +143,10 @@ template <typename CONFIG> class ExtensionField {
     // inverse assumes that xs is nonzero
     static constexpr HOST_DEVICE_INLINE ExtensionField inverse(const ExtensionField& xs) {
       ExtensionField xs_conjugate = { xs.real, FF::neg(xs.imaginary) };
+      FF i_sq_times_im = FF::template mul_unsigned<CONFIG::i_squared>(FF::sqr(xs.imaginary));
+      i_sq_times_im = CONFIG::i_squared_is_negative ? FF::neg(i_sq_times_im) : i_sq_times_im;
       // TODO: wide here
```

Comment:
well this todo is not yet fully resolved (it reminds me that adding up two products can be done with 1 mod reduction instead of 2) so not gonna remove it yet